docs/content/core_concept.md

```diff
@@ -141,80 +143,118 @@ The `IP` object has two purposes:
+| url                     | URL to fetch IPv4 address                                                                                                                                            | `https://ipecho.net`                  |
+| requestBody             | Additional request body to send to the `url`                                                                                                                         |                                       |
+| requestHeaders          | Additional request headers to send to the `url`. The key will be passed as http header and the value will be passed as headers value                                 | `Accept: application/json`            |
+| requestHeadersSecretRef | link to a `secret` with additional http headers. All secret's key will be passed as http header and the corresponding secret's value will be passed as headers value | see example below                     |
+| requestMethod           | request method. Possible values are `GET`, `POST`, `PUT` or `DELETE`                                                                                                 | `GET`                                 |
+| responseTextRegex       | If the IPv4 address must be extracted from the http response. Uses the default golang regex engine.                                                                  | `\d{1,3}\.\d{1,3}.\.\d{1,3}\.\d{1,3}` |
+| responseJSONPath        | jsonpath to extract Ipv4 address. Uses the kubectl jsonpath library                                                                                                  | `'{.ip}'`                             |
 
-    !!! warning
-        The source must return only the external IPv4 address.
+!!! warning "responseTextRegex"
+    Be aware that the http request will fetch the **complete html document** and not what you see in your browser!
 
-        Good example:
+!!! note
+    If neither `responseJSONPath` nor `responseTextRegex` is set, cloudflare-operator will try to parse the **complete html document** as IPv4 address.
 
-        ```bash
-        curl https://api.ipify.org
-        ```
+Examples:
 
-        Output:
+**Fetch your external IPv4 address from three "IP" providers**
 
-        ```console
-        142.251.36.35
-        ```
+```yaml
+---
+apiVersion: cf.containeroo.ch/v1beta1
+kind: IP
+metadata:
+  name: external-ipv4
+spec:
+  type: dynamic
+  IpSources:
+    - url: https://ipecho.net
+      responseTextRegex: \d{1,3}\.\d{1,3}.\.\d{1,3}\.\d{1,3}
+    - url: https://api.ipify.org?format=json
+      responseJSONPath: '{.ip}'
+    - url: https://checkip.amazonaws.com
+  interval: 5m
+```
 
-        Bad example:
+!!! info
+    Because more than one `source` is set, cloudflare-operator shuffle the list with `IpSources` and try to fetch a valid IPv4 address, until the first response is valid. If none of the `IpSources` return a valid IPv4 address, cloudflare-operator will set the status of the `IP` object as `failed`.
```

Comment:
```suggestion
    Because more than one `source` is set, cloudflare-operator shuffles the list with `ipSources` and tries to fetch a valid IPv4 address, until the response is valid. If none of the `ipSources` returns a valid IPv4 address, cloudflare-operator will set the status of the `IP` object to `failed`.
```